docx/enum/header.py

```diff
@@ -0,0 +1,39 @@
+# encoding: utf-8
+
+from __future__ import (
+    absolute_import, print_function, unicode_literals, division
+)
+
+from .base import alias, XmlEnumeration, XmlMappedEnumMember
+
+
+@alias('WD_HEADER_TYPE')
+class WD_HEADER_FOOTER_INDEX(XmlEnumeration):
+    """
+    alias: **WD_HEADER_TYPE**
+
+    Specifies the type of a header.
+
+    Example::
+
+        from docx.enum.header import WD_HEADER_TYPE
+
+        header = section.header
+        header.type = WD_HEADER_TYPE.PRIMARY
+    """
+
+    __ms_name__ = 'WdHeaderFooterIndex'
+
+    __url__ = 'https://msdn.microsoft.com/en-us/library/office/ff839314.aspx'
+
+    __members__ = (
+        XmlMappedEnumMember(
+            'PRIMARY', 1, 'default', 'Default/odd header type.'
+        ),
+        XmlMappedEnumMember(
+            'FIRST_PAGE', 2, 'first', 'First page header type.'
+        ),
+        XmlMappedEnumMember(
+            'EVEN_PAGES', 3, 'even', 'Even page header type.'
+        ),
+    )
```

Comment:
These are merged, these comments are for future reference to account for the changes I made during merge.

* This one is only used internally. As far as I know there is no public API that uses this and there won't be a `.type` property on Header, so I removed the example.
* The alias, if there is one, is the part of the MS member names that's the same for each member. In this case it's WdHeaderFooter -> WD_HEADER_FOOTER
* Enum modules get a docstring, as does every module.
* The description string for each member is generally copied directly from the MS API web page, adapted only when necessary. This one requires quite a bit of interpolation since it's used a bit differently and the MS ones are misleading and in cases just plain wrong :)
* The enumeration documentation page is generated automatically from the enumeration definition like this:

  ```python
  >>> from docx.enum.header import WD_HEADER_FOOTER as WD
  >>> print WD.__docs_rst__
```

```rst
.. _WdHeaderFooterIndex:

``WD_HEADER_FOOTER_INDEX``
==========================

alias: **WD_HEADER_FOOTER**

Specifies the type of a header or footer.

----

PRIMARY
    The header or footer appearing on all pages except when an even and/or
    first-page header/footer is defined.

FIRST_PAGE
    The header or footer appearing only on the first page of the specified
    section.

EVEN_PAGES
    The header or footer appearing on even numbered (verso) pages in the
    specified section.
```

This text is used to create the documentation page you can see in the merged commit (along with it's index entry).